DS-3693: Add plugin to index filenames and file descriptions for files in ORIGINAL bundle

MW3000 opened the pull request:
Hi. I wrote a tiny plugin, `SolrServiceFileInfoPlugin`, (implementing `SolrServiceIndexPlugin`) to make filenames and file descriptions from the ORIGINAL bundle searchable.

The plugin must be added as a `bean` to `discovery.xml`.
To see it's effect, you have to reindex the Solr search core by executing:
`[dspace]/bin/dspace index-discovery -b`

I documented it on https://gist.github.com/MW3000/5771c0488f8952bb888357ce8c655279

Andrea Schweer and Terry Brady encouraged me to make a pull request.

Commit:
Add plugin to index filenames and file descriptions for files in ORIGINAL bundle

MW3000 commented:
I also opened a ticket: https://jira.duraspace.org/browse/DS-3693

Commit:
Add expected license header

tomdesair commented:
Can you also add a search filter for this in `discovery.xml` so that these fields can be queried from the web ui?

Commit:
Add search filters

- add search filter for original_bundle_filenames and original_bundle_descriptions to discovery.xml
-  add messages

MW3000 commented:
@tomdesair That's a good idea. I added the search filters.

MW3000 commented:
Travis is not running, instead it says on https://travis-ci.org/DSpace/DSpace/requests :

>  #1838 d9ce088 Abuse detected

Can someone fix this?

Commit:
Empty commit to trigger Travis CI

terrywbrady reviewed (approved):
Code changes look good.  Tested locally, and the new filters work as expected.

![image](https://user-images.githubusercontent.com/1111057/37434115-bd02f192-279b-11e8-8d48-e2f5b4bb634e.png)

Is there any reason that an institution would mind having this filter on by default?

kshepherd commented:
this looks great and I'm +1 on adding it as is. thanks @MW3000 !
i'm trying to think of how this interacts with embargos / bitstreams that institutions have deliberately hidden in UI (based on metadata or auth policy), even if they're in ORIGINAL bundle... that's one reason why I think it might be nice to not enable by default just yet, add some documentation for it and do a bit of testing with restricted/embargoed bitstreams and look to enable by default in 6.4?
ping: @tomdesair , @MW3000 for your thoughts

kshepherd commented:
@tomdesair : we're a +1 off being able to merge this, could you take another look now that the requested changes are made and travis is happy, and give your approval?

kshepherd commented:
I'm merging this now as it doesn't look like @tomdesair is able to review, and we've had 2 +1s anyway, for this small and low-impact PR I'm going to just press the button!
Thanks @MW3000 . Just a note that we'll need to port this to master branch as well, should be pretty easy, we just won't want the changes to messages.xml (as xmlui isn't in master)

MW3000 commented:
I'll see if i find time to port it to master next week.

MW3000 commented:
I ported it to master in PR #2059
